lgn-provers/src/provers/v0/query/erc20/prover.rs

```diff
@@ -0,0 +1,260 @@
+use crate::params::ParamsLoader;
+use ethers::addressbook::Address;
+use lgn_messages::types::v0::query::erc20::{RevelationData, StorageBranchInput, StorageLeafInput};
+use lgn_messages::types::v0::query::{PartialNodeBlockData, QueryStateData};
+use mr_plonky2_circuits::api::{QueryInput, QueryParameters};
+use mr_plonky2_circuits::query_erc20;
+use mr_plonky2_circuits::query_erc20::RevelationErcInput;
+use tracing::{debug, info};
+
+pub trait QueryProver {
+    fn prove_storage_leaf(&self, data: &StorageLeafInput) -> anyhow::Result<Vec<u8>>;
+
+    fn prove_storage_branch(&self, data: &StorageBranchInput) -> anyhow::Result<Vec<u8>>;
+
+    fn prove_state_db(&self, contract: Address, data: &QueryStateData) -> anyhow::Result<Vec<u8>>;
+
+    fn prove_block_partial_node(&self, data: &PartialNodeBlockData) -> anyhow::Result<Vec<u8>>;
+
+    fn prove_block_full_node(
+        &self,
+        left_proof: &[u8],
+        right_proof: &[u8],
+    ) -> anyhow::Result<Vec<u8>>;
+
+    fn prove_revelation(&self, data: &RevelationData) -> anyhow::Result<Vec<u8>>;
+}
+
+pub struct EuclidProver {
+    params: QueryParameters<21, 5>,
```

Comment:
Can't we include that in constants so we are sure these are the same between query2 and queryerc ?